### ISSUE TITLE
What is MLTable able to filter?

### ISSUE THREAD

**Issue by TomSB1423** [Azure/azure-sdk-for-python#30821]
The [documentation](https://learn.microsoft.com/en-us/python/api/mltable/mltable.mltable.mltable?view=azure-ml-py#mltable-mltable-mltable-filter) states that MLTable is able to filter using an "expression". What types of expressions are available to use?

I would like to be able to filter by datetime object e.g col(date) > 30/01/2023, however it seems like if I write this condition in the filter, MLTable does something else like a string → int conversion and then compares int. The filter does not work correctly for dates.

Would you be able to suggest an alternative approach if not?

**Comment by pvaneck** [comment]
Hey, thanks for the feedback. @azureml-github, can some guidance be provided for filtering?

**Comment by xiongrenyi** [comment]
Can you try this? mltable.filter('Date >= datetime(2015, 7, 5, 23)')
We'll add valid 'expression' in details to the document shortly

**Comment by TomSB1423** [comment]
Any update on this?

**Comment by samuel100** [comment]
@TomSB1423 you need to use `col()` i.e.

```python
tbl = mltable.load(path)
tbl = tbl.filter("col('TimeFeature') <= datetime(2022, 05, 16)")
```

`filter()` function follows Python expressions: https://docs.python.org/3/reference/expressions.html#

**Comment by samuel100** [comment]
#please-close

**Comment by TomSB1423** [comment]
Note: For future use - datetime objects must be in the format datetime(year, month, day) with month and day **not** containing 0 as a prefix. E.g. first of the month should be represented as "1" not "01". 
Example: datetime(2023, 1, 5) not datetime(2023, 01, 05)

**Comment by lucasfijen** [comment]
/unresolve I have actually a very similar question, to me it seems like only a very limited set of operators in these expressions is allowed, would it be an idea to update the documentation on this?

I'm trying to compare my column to a list of values I want to filter out of my mltable. In normal python expressions I would use the 'in' operator with a list comparison, so say for example `col("name") in ["a", "b", "c"]`. However this doesn't seem to be allowed syntax in the filter function, am I right to understand that my only way to achieve this would be to perform a `(col("name") == "a") or (col("name") == "c")`? 

-sorry if this is hijacking the thread, but I felt like I was experiencing the same issue/question: The desire to have a better understanding of what exact python expressions are and are not allowed in this filter field

**Comment by TomSB1423** [comment]
@lucasfijen I also agree that there should be better documentation for MLTable in general.
1. What is MLTable able to filter - "expression" is just not detailed enough. [Link](https://learn.microsoft.com/en-us/python/api/mltable/mltable.mltable.mltable?view=azure-ml-py#mltable-mltable-mltable-filter)
2. convert_column_types appears to have missing information - [link](https://learn.microsoft.com/en-us/python/api/mltable/mltable.mltable.mltable?view=azure-ml-py#mltable-mltable-mltable-convert-column-types):
<img width="461" alt="image" src="https://github.com/Azure/azure-sdk-for-python/assets/63239252/8b2f6aa2-2724-4bb8-9bed-623f3e295e0a">

**Comment by chanccm** [comment]
I'm trying to filter a datetime column by yyyy-mm-dd:
tbl = tbl.filter("col('Sampling_DateTime') <= date_one_week_ago")

where 'date_one_week_ago' is a variable that changes every time I run the script.

What is the proper way to enter the variable 'date_one_week_ago' into the tbl.filter() please?
